src/job_queue/queue.rs

```diff
@@ -102,11 +102,17 @@ impl<P: Ord + Send + Sync + 'static> JobQueue<P> {
         Ok(())
     }
 
-    /// adds job to job-queue and returns immediately.
+    /// adds job to job-queue (with interior mutability)
     ///
-    /// job-results can be obtained by via JobHandle::results().await
-    /// The job can be cancelled by JobHandle::cancel()
-    pub fn add_job(&self, job: Box<dyn Job>, priority: P) -> Result<JobHandle, AddJobError> {
+    /// returns immediately (does not wait for job).
```

Comment:
“does not wait for job [completion]”?

Reply:
addressed in 8b189964d2c0fd51b8e9c4f7899b2f713bc95b99